osu.Game/Overlays/Music/PlaylistItem.cs

```diff
@@ -109,7 +109,7 @@ public bool InSelectedCollection
             }
         }
 
-        public IEnumerable<string> FilterTerms { get; }
+        public IEnumerable<string> FilterTerms => Model.Metadata.GetSearchableTerms();
```

Comment:
So this I find a little risky, because in real non-test usages this `Model` isn't detached anymore (detach was removed when moving the subscription from the music controller to the playlist). So I believe that because the model is managed, this property is essentially only safe to evaluate from the update thread by extension - which in this case, it happens to be accessed correctly by the virtue of how `RearrangeableListItemContainer` happens to work. But `FilterTerms` itself looks to be just data and not something you'd need to be careful with wrt threading.

Not sure whether this is something that needs addressing. Given that this component in general doesn't support on-the-fly updates it may be better to copy the filter terms out in `LoadComplete()`. Or change this to a `Live<T>`. Or just disregard completely and hope it doesn't break, I dunno.

Reply:
Right, I actually picked up on this while working on it but forgot to resolve. I think an `ILive` will work well here.

Reply:
Has been updated to use `ILive`. A bit ugly, but safer I guess.